src/main/takeDetection/takeDetection.ts

```diff
@@ -0,0 +1,104 @@
+import { Word, IndexRange } from 'sharedTypes';
+import {
+  InjectableTake,
+  InjectableTakeGroup,
+} from 'main/editDelete/injectTakeInfo';
+import getSimilarityScore from './sentenceSimilarity';
+
+export type Sentence = {
+  startIndex: number | null;
+  endIndex: number | null;
+  sentenceString: string;
+};
+
+export function findSentences(words: Word[]): Sentence[] {
+  let currentSentence: Sentence = {
+    startIndex: 0,
+    endIndex: null,
+    sentenceString: '',
+  };
+  const sentences: Sentence[] = [];
+  words.forEach((word, idx) => {
+    if (currentSentence.sentenceString === '') {
+      currentSentence.sentenceString = word.word;
+    } else {
+      currentSentence.sentenceString = currentSentence.sentenceString.concat(
+        ' ',
+        word.word.toString()
+      );
+    }
+
+    if (
+      word.word.includes('.') ||
+      word.word.includes('?') ||
+      word.word.includes('!') ||
+      idx === words.length - 1
+    ) {
+      currentSentence.endIndex = idx + 1;
+      sentences.push(currentSentence);
+      currentSentence = {
+        startIndex: idx + 1,
+        endIndex: null,
+        sentenceString: '',
+      };
+    }
+  });
+  sentences[sentences.length - 1].endIndex = words.length;
+  return sentences;
+}
+
+function newTakeGroup(
+  sentenceOne: Sentence,
+  sentenceTwo: Sentence
+): InjectableTakeGroup {
+  const newGroup: InjectableTakeGroup = { takes: [] };
+  let nextTakeRange: IndexRange = {
+    startIndex: sentenceOne.startIndex as number,
+    endIndex: sentenceOne.endIndex as number,
+  };
+  let nextTake: InjectableTake = { wordRange: nextTakeRange };
+  newGroup.takes.push(nextTake);
+  nextTakeRange = {
+    startIndex: sentenceTwo.startIndex as number,
+    endIndex: sentenceTwo.endIndex as number,
+  };
+  nextTake = { wordRange: nextTakeRange };
+  newGroup.takes.push(nextTake);
+  return newGroup;
+}
+
+export function findTakes(words: Word[]): InjectableTakeGroup[] {
+  const takeGroups: InjectableTakeGroup[] = [];
+  const sentences: Sentence[] = findSentences(words);
+  for (let i = 0; i < sentences.length - 1; i += 1) {
+    const similarity: number = getSimilarityScore(
+      sentences[i].sentenceString,
```

Comment:
You reference `sentences[i]` and `sentences[i+1]` a lot, give them meaningful names e.g. `currentSentence` and `nextSentence`